### PR TITLE
[aes] Add list of countermeasures in preparation for D2S sign-off

### PR DESCRIPTION
**Pull request by vogelpi**
This PR adds the list of countermeasures for AES in preparation for the D2S sign-off.

This requires #9816 to go in first.

### CODE REVIEW

File: hw/ip/aes/data/aes.hjson
Comment (vogelpi) [inline hw/ip/aes/data/aes.hjson:227]: I wasn't sure whether to put `KEY` or `DATA_REG` here. `DATA_REG` stands for "non-software config data registers (e.g. keccak state)". Could also list the two separately. WDYT?

File: hw/ip/aes/doc/checklist.md
Comment (vogelpi) [inline hw/ip/aes/doc/checklist.md:78]: There is still some review backlog for other AES changes. At the same, datapath FI hardening has never been a strict requirement for the first version. We've thus decided to take AES to D2S before datapath FI hardening is complete and continue the implementation in parallel.

File: hw/ip/aes/data/aes.hjson
Comment (rswarbrick) [inline hw/ip/aes/data/aes.hjson:280]: Could you add a note here to explain why this is included but commented out? ("We're gonna do this, but haven't done it yet" or similar)

File: util/reggen/countermeasure.py
Comment (rswarbrick) [inline util/reggen/countermeasure.py:34]: Missing comma here!

File: util/reggen/countermeasure.py
Comment (rswarbrick) [inline util/reggen/countermeasure.py:44]: Missing comma here!

**Review by rswarbrick — COMMENTED** [review]
The failing CI is because of the missing commas in `countermeasure.py`

File: util/reggen/countermeasure.py
Comment (rswarbrick) [reply util/reggen/countermeasure.py]: Also, I think the only documentation for these names is in the (confidential) countermeasures sheet: is that right? If so, we should probably keep that in sync with this file.

File: hw/ip/aes/data/aes.hjson
Comment (rswarbrick) [reply hw/ip/aes/data/aes.hjson:227]: I guess maybe I'd split it, but don't have strong feelings.

File: hw/ip/aes/data/aes.hjson
Comment (rswarbrick) [inline hw/ip/aes/data/aes.hjson:231]: Are there situations where they're cleared with something else? Otherwise, "are cleared" might be better?

File: hw/ip/aes/data/aes.hjson
Comment (vogelpi) [reply hw/ip/aes/data/aes.hjson:282]: Done.

File: hw/ip/aes/data/aes.hjson
Comment (rswarbrick) [inline hw/ip/aes/data/aes.hjson:233]: I'm not sure that this is how the countermeasure instance name is supposed to be used. @msfschaffner: Is this supposed to be allowed? If not, I think there's a missing check in the Python :-)

That said, I think it's a good idea! If it's not currently how we intend things to work, maybe we should explicitly allow it? (In which case, I'd suggest allowing hierarchical instance names, so use `rsplit(..., 2)` in the Python and treat the stuff to the left as an opaque blob)

File: util/reggen/countermeasure.py
Comment (vogelpi) [reply util/reggen/countermeasure.py]: Thanks for spotting this. I've fixed that now.

File: hw/ip/aes/data/aes.hjson
Comment (rswarbrick) [inline hw/ip/aes/data/aes.hjson:264]: Nit: This would probably read slightly more easily with "... counter that counts in ... "

File: util/reggen/countermeasure.py
Comment (rswarbrick) [reply util/reggen/countermeasure.py]: About the documentation, I've also just opened https://github.com/lowRISC/opentitan/issues/9985, which tracks documenting these properly.

File: hw/ip/aes/data/aes.hjson
Comment (vogelpi) [reply hw/ip/aes/data/aes.hjson:230]: Sounds good, I've now split it up into to items for `KEY` and `DATA_REG`.

File: hw/ip/aes/data/aes.hjson
Comment (vogelpi) [reply hw/ip/aes/data/aes.hjson:239]: `MAIN` is the optional instance name and `FSM` the asset. This is inline with what @msfschaffner proposed in #9816.

File: hw/ip/aes/data/aes.hjson
Comment (vogelpi) [reply hw/ip/aes/data/aes.hjson]: Yep, I agree.

**Comment by vogelpi** [comment]
Thanks @rswarbrick for your review.

Now CI is failing because I didn't rerun reggen. But this is only possible after merging #9816 .

File: doc/rm/comportability_specification/index.md
Comment (msfschaffner) [inline doc/rm/comportability_specification/index.md:303]: this one makes sense to me

File: doc/rm/comportability_specification/index.md
Comment (msfschaffner) [inline doc/rm/comportability_specification/index.md:302]: isn't this similar to `SW_UNREADABLE`?

File: hw/ip/aes/data/aes.hjson
Comment (msfschaffner) [reply hw/ip/aes/data/aes.hjson:245]: So the naming convention we outlined was to allow for an optional uniquifier in front of the canonical name, see also on the spreadsheet where the naming convention has initially been proposed.

I am not sure we should go full-on hierarchical with unbounded number uf hierarchy levels, though.
My concern is that this may just lead to more naming discussions, as it would allow for more naming combinations.

Comment (msfschaffner) [reply hw/ip/aes/data/aes.hjson:245]: @tjaychen

Comment (tjaychen) [reply hw/ip/aes/data/aes.hjson:245]: yeah the optional prefix was mostly a mechanism to allow the countermeasure writer a way to differentiate among instances of countermeasures.  There are not a lot of strict rules about this.  I would prefer not to go down the path of full hierarchy, because if we do, the question will become how we validate that.

**Comment by vogelpi** [comment]
Update: rebased on top of master and reworked the labels as discussed in the D2S review meeting. In particular:
- Label IV as seperate asset (IV.CONFIG.SW_UNREADABLE and IV.CONFIG.SEC_WIPE)
- Expand description of all FSM.LOCAL_ESC to indicate what can trigger the escalation. Related to that, remove non-FSM-specific LOCAL_ESC labels. One exemption is DATA_REG.LOCAL_ESC: I think it makes sense to indicate that we don't release intermediate data upon escalation.
- Make KEY.SCA more specific (DATA_REG.KEY.SCA).
- Add LC_ESCALATE_EN.INTERSIG.MUBI label.

**Review by msfschaffner — APPROVED** [review]
LGTM, thanks for updating this.

**Comment by cdgori** [comment]
When this lands we (I) have to update the spreadsheet, I think?

**Comment by vogelpi** [comment]
> When this lands we (I) have to update the spreadsheet, I think?

Yes, this is my understanding, too. I don't have write access to the spreadsheet.

**Comment by cdgori** [comment]
> > When this lands we (I) have to update the spreadsheet, I think?
> 
> Yes, this is my understanding, too. I don't have write access to the spreadsheet.

OK, I have at least gotten everything from the HJSON into the spreadsheet and collapsed/expanded entries where I could.  I need some help (from you, I think?) to resolve the ones at the bottom of the AES section that link to #1005 (don't worry about the REJECTED ones, they will carry over fine).  I'm not sure if: they are still to-be-done (or consciously not-to-be-done and thus converted to REJECTED), could be merged into other entries, or require another countermeasure entry in the HJSON.  If you have time after tomorrow's Si meeting we might be able to finish it off then?

**Pull request merged**
